unity-renderer/Assets/DCLServices/MapRendererV2/MapLayers/SatelliteAtlas/SatelliteChunkController.cs

```diff
@@ -0,0 +1,96 @@
+using UnityEngine.Networking;
+using Object = UnityEngine.Object;
+
+namespace DCLServices.MapRendererV2.MapLayers.SatelliteAtlas
+{
+    public class SatelliteChunkController : IChunkController
+    {
+        private const string CHUNKS_API = "https://media.githubusercontent.com/media/genesis-city/genesis.city/master/map/latest/3/";
+
+        private readonly SpriteRenderer spriteRenderer;
+
+        private Service<IWebRequestController> webRequestController;
+
+        private CancellationTokenSource internalCts;
+        private CancellationTokenSource linkedCts;
+        private int webRequestAttempts;
+
+        public SatelliteChunkController(SpriteRenderer prefab, Vector3 chunkLocalPosition, Vector2Int coordsCenter, Transform parent, int drawOrder)
+        {
+            internalCts = new CancellationTokenSource();
+
+            spriteRenderer = Object.Instantiate(prefab, parent);
+            spriteRenderer.transform.localPosition = chunkLocalPosition;
+            spriteRenderer.sortingOrder = drawOrder;
+
+#if UNITY_EDITOR
+            spriteRenderer.gameObject.name = $"Chunk {coordsCenter.x},{coordsCenter.y}";
+#endif
+        }
+
+        public void Dispose()
+        {
+            internalCts?.Cancel();
+            linkedCts?.Dispose();
+            linkedCts = null;
+
+            internalCts?.Dispose();
+            internalCts = null;
+
+            if (spriteRenderer)
+                Utils.SafeDestroy(spriteRenderer.gameObject);
+        }
+
+        public async UniTask LoadImage(Vector2Int chunkId, float chunkWorldSize, CancellationToken ct)
+        {
+            webRequestAttempts = 0;
+            linkedCts = CancellationTokenSource.CreateLinkedTokenSource(internalCts.Token, ct);
+
+            var url = $"{CHUNKS_API}{chunkId.x}%2C{chunkId.y}.jpg";
+            UnityWebRequest webRequest = await GetTextureFromWebAsync(url, ct: linkedCts.Token);
+
+            Texture2D texture = CreateTexture(webRequest.downloadHandler.data);
+            texture.wrapMode = TextureWrapMode.Clamp;
+
+            float pixelsPerUnit = texture.width / chunkWorldSize;
+
+            spriteRenderer.sprite = Sprite.Create(texture, new Rect(0, 0, texture.width, texture.height), Vector2Ext.OneHalf, pixelsPerUnit,
+                0, SpriteMeshType.FullRect, Vector4.one, false);
+
+            return;
+            Texture2D CreateTexture(byte[] data)
+            {
+                var texture2D = new Texture2D(1, 1);
+                texture2D.LoadImage(data);
+                return texture2D;
+            }
+        }
+
+        private async UniTask<UnityWebRequest> GetTextureFromWebAsync(string url, int maxAttempts = 3, CancellationToken ct = default)
+        {
+            UnityWebRequest webRequest;
+
+            try { webRequest = await webRequestController.Ref.GetTextureAsync(url, cancellationToken: linkedCts.Token); }
+            catch (Exception e)
+            {
+                if (webRequestAttempts < maxAttempts)
```

Comment:
It seems redundant as repetition logic is already in ` webRequestController.Ref.GetTextureAsync`. This way you repeat the request 9 times.

Reply:
I actually tried it with repetition logic from `webRequestController.Ref.GetTextureAsync` but it didn't fix the SSL connection exception I was getting in Editor. I can make branching here, so this part will be Editor-only. What do you think?